Adding code of conduct

ram-cherukuri opened the pull request:
Adopting the code of conduct from the https://www.contributor-covenant.org/

<!-- markdownlint-disable MD013-->
# PhysicsNeMo Pull Request

## Description
<!-- Provide a standalone description of changes in this PR. -->
<!-- Reference any issues closed by this PR with "closes #1234". -->
<!-- Note: The pull request title will be included in the CHANGELOG. -->

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

## Dependencies

<!-- Call out any new dependencies needed if any -->

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

ram-cherukuri committed:
Adding code of conduct

Adopting the code of conduct from the https://www.contributor-covenant.org/

mnabian committed:
Update CODE_OF_CONDUCT.MD

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

mnabian reviewed (approved):
LGTM. Fixed a typo in the PhysicsNeMo team email address.

coreyjadams reviewed (approved):
I like that we're adding this. Thanks @ram-cherukuri !

coreyjadams committed:
Create .markdownlintignore

mnabian committed:
Merge branch 'main' into ram-cherukuri-patch-2

mnabian commented:
/blossom-ci

ram-cherukuri committed:
Revise README for PhysicsNeMo resources and guidance

Updated the 'Getting Started' section and added new resources for learning AI Physics.

coreyjadams committed:
Merge branch 'main' into ram-cherukuri-patch-2

coreyjadams committed:
Merge branch 'main' into ram-cherukuri-patch-2

coreyjadams committed:
Update README.md

coreyjadams committed:
Merge branch 'main' into ram-cherukuri-patch-2

coreyjadams commented:
/blossom-ci
